### PR TITLE
Add introduction section to documentation 

### PR DESCRIPTION
**Pull request by Hombre-x**
Add a quasi-faq section to the documentation talking about the imports, API usage and a Whats this IO thing section speaking of the advantages of using it (only referential transparency so far, so more suggestions are welcome!!!!)

Also, more introduction related topic suggestions don't hurt ;)

### CODE REVIEW

File: docs/introduction/index.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# First steps
+
+## Which API should I use?
+
+In shellfish, you have two different ways of doing things:
+
+- First, if you prefer a more concise, fluent syntax, use the extension methods (e.g. `path.read()`) by importing the `shellfish.syntax.all.*` package.
```
Comment (TonioGela) [inline docs/introduction/index.md]: Is the import correct?

```suggestion
- First, if you prefer a more concise syntax, use the extension methods (e.g. `path.read()`) by importing the `shellfish.syntax.all.*` package.
```

File: docs/introduction/index.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+- If you prefer calling static methods, use direct method calls on the `FilesOs` object (e.g., `FilesOs.read(path)`). You can also import all the functions inside the `shellfish.FilesOs` package if you don't want to call the `FilesOs` object every time (e.g., `read(path)`). 
+
+    For this documentation we'll call the methods on the `FilesOs` objects in the static variant to difference them from the extension methods.
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
In this documentation we'll call the methods on the `FilesOs` objects in the static variant to differentiate them from the extension ones.
```

File: docs/introduction/index.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+for
+  file <- FilesOs.read(path)
+  _    <- FilesOs.append(path, "I'll place this here.")
+yield ()
+```
+
+@:@
+
+Which one you should use really depends on your preferences and choices; if you like the method style of calling functions directly to the object, stick to the extension methods syntax, if you rather prefer a more Haskell-like style, stick to the static object calls!
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
Which one you should use really depends on your preferences and choices; if you like the method style of calling functions directly on the objects, stick to the extension methods syntax, if you rather prefer a more Haskell-like style, stick to the static object calls!
```

File: docs/introduction/index.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+## Imports
+
+If you just want to start scripting right away, importing `shellfish.*` will do the trick; it imports all the extension methods and functionality you need, such as types and functions, to start working right away.
+
+But if you want more concise functionality, the `shellfish.syntax.path.*` and `shellfish.syntax.process.*` will only import the extension methods.
+
+For the static methods, the `shellfish.FilesOs` and `shellfish.ProcessesOs` imports will provide the functions to work with files and processes, if that is your preferred style.
+
+Finally, `shellfish.domain.*` will import all the types needed to work with the library!
```
Comment (TonioGela) [inline docs/introduction/index.md]: I didn't see all this imports coming :P
In any case I'll say that we can completely skip this section, and let the users do everything with a single `shellfish.*` import

File: docs/introduction/index.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+To actually run the computation, you have two options, the first one (and not recommended) is to call the `unsafeRunSync()` function at the very end of the program:
+
+```scala mdoc
+import cats.effect.unsafe.implicits.global // Imports the runtime that executes the IO monad
+
+suspendingHello.unsafeRunSync()
+```
+
+But this is not the usual way; the common way is to pass it to the `run` function (similar to the main method in scala, but for `IO`!). To that, you have to extend the object with `IOApp`:
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
But this is not the usual way: the usual way is to assing it to the `run` function (similar to the main method but for `IO`). To that, you have to extend your application's main object with `IOApp`:
```

File: docs/introduction/index.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+```scala mdoc:compile-only
+import cats.syntax.all.*
+
+val rndNumIO = IO(Random.nextInt(10))
+
+val result1IO = (rndNumIO, rndNumIO).mapN(_ + _)
+
+val result2IO = (IO(Random.nextInt(10)), IO(Random.nextInt(10))).mapN(_ + _)
+```
```
Comment (TonioGela) [inline docs/introduction/index.md]: Can I ask you to rework this and not use `Random` as an example? You're correctly claiming that `result1 != result2` in the first example, but you are not claiming that the net result of running the second is ``result1 == result2`, because you've picked `Random` that is random and as such would be unpredicable.

Also, don't use `mapN` it will be confusing for newcomers, use a for comprehension.

Use maybe `val salute = println("Hello")` to prove that `salute + salute` won't fire two times.

File: docs/introduction/index.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+For more examples of why this is important, check out [this blog post](https://blog.rockthejvm.com/referential-transparency/) about referential transparency!
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
Here's a [good explanation](https://blog.rockthejvm.com/referential-transparency/) about the benefits of referential transparency.
```

File: docs/introduction/index.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+
+Another advantage is control over the code. Because "wrapping" computations inside the `IO` monad converts programs into descriptions of programs instead of statements, you automatically gain control over your program as to when and when not to execute those statements.
+
+For those and many more reasons, we opted to make this library pure by implementing `IO` on the library!
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
```

File: docs/introduction/index.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+val result1IO = (rndNumIO, rndNumIO).mapN(_ + _)
+
+val result2IO = (IO(Random.nextInt(10)), IO(Random.nextInt(10))).mapN(_ + _)
+```
+
+For more examples of why this is important, check out [this blog post](https://blog.rockthejvm.com/referential-transparency/) about referential transparency!
+
+
+Another advantage is control over the code. Because "wrapping" computations inside the `IO` monad converts programs into descriptions of programs instead of statements, you automatically gain control over your program as to when and when not to execute those statements.
```
Comment (TonioGela) [inline docs/introduction/index.md]: That's terse, can you rephrase it?

**Review by TonioGela — CHANGES_REQUESTED** [review]
Few changes required

File: docs/introduction/index.md (hunk 5 above)
Comment (zetashift) [reply docs/introduction/index.md]: passing* :P

File: docs/introduction/index.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+import cats.effect.IOApp
+
+object Main extends IOApp.Simple:
+
+  def run: IO[Unit] = suspendingHello
+
+end Main
+```
+
+On either way, the `IO` will be executed and all the computation will be evaluated.
```
Comment (zetashift) [inline docs/introduction/index.md]:
```suggestion
Either way, the `IO` will be executed and all the computation will be evaluated.
```

File: docs/introduction/index.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+import cats.effect.IO
+
+/* Will print to de console! */
+val printingHello = println("Hello newbies!")
+
+/* Will not do anything (yet) */
+val suspendingHello = IO(println("Hello newbies!"))
+```
+
+To actually run the computation, you have two options, the first one (and not recommended) is to call the `unsafeRunSync()` function at the very end of the program:
```
Comment (zetashift) [inline docs/introduction/index.md:88]: If this is doc is for beginners, I'd recommend replacing `computation` with `program`.
Or a small explanation what a `computation` is.

**Review by zetashift — CHANGES_REQUESTED** [review]

File: docs/introduction/index.md (hunk 1 above)
Comment (Hombre-x) [reply docs/introduction/index.md]: Not correct, but I made It when I was thinking about the Processes API as well, I'll change it to `syntax.path.*` :)

File: docs/introduction/index.md (hunk 6 above)
Comment (Hombre-x) [reply docs/introduction/index.md]: Yup! Let me see what can I do :D

File: docs/introduction/index.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+```
+Now both results are the same, an behave exactly the same!
+
+Here's a [good explanation](https://blog.rockthejvm.com/referential-transparency/) about the benefits of referential transparency.  
+
+
+Another benefit is gaining explicit control over code execution. By encapsulating computations within the `IO` monad, your programs become blueprints rather than direct statements. This gives you the ability to decide precisely when to execute those statements.
+
+
+## Weird `>>` and `>>=` methods, what are these?
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
## Weird `>>` and `>>=` operators, what are those?
```

File: docs/introduction/index.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+
+While reading the documentation of this library, you may come across some strange like `>>`, don't worry, this is convenience syntax sugar for some fairly common methods!
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
While reading the documentation of this library, you may come across some strange operator like `>>`. This is convenient syntax sugar for some fairly common methods!
```

File: docs/introduction/index.md [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+```scala mdoc
+import cats.syntax.all.*
+```
+
+For instance, you may seen something like this:
+
+```scala mdoc:compile-only
+IO("The result is: 42") >>= IO.println
+```
+That is just an alias for `flatMap`, it's like writing `IO("The result is: 42").flatMap(IO.println(_))`, but without the added boilerplate. This use is more common in languages like Haskell, but we'll use it in the documentation to simplify things a bit!
```
Comment (TonioGela) [inline docs/introduction/index.md]:
```suggestion
That is just an alias for `flatMap`, so it's like writing `IO("The result is: 42").flatMap(IO.println(_))`, but without the added boilerplate. This use is more common in languages like Haskell, but we'll use it in the documentation to simplify things a bit!
```

**Review by TonioGela — CHANGES_REQUESTED** [review]
Add just the two tiny changes I suggested you and we're good to go.
